### PR TITLE
refactor(ci): to use github app auth over long-lived credentials

### PR DESCRIPTION
**Pull request by tdeekens**
#### Summary

This refactors the authentication for the GitHub PAT for releasing via Changesets from a long-lived static credential to a short-lived dynamic credential issues through a GitHub App.

More information can be found in internal documentation or repositories.

### CODE REVIEW

**Review by CarlosCortizasCT — APPROVED** [review]
Thanks!

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -12,14 +12,22 @@ jobs:
...
+        with:
+          app_id: ${{ secrets.CT_CHANGESETS_APP_ID }}
+          private_key: ${{ secrets.CT_CHANGESETS_APP_PEM }}
+
       - name: Checkout
         uses: actions/checkout@v3
         with:
-          # Pass a personal access token (using our `ct-release-bot` account) to be able to trigger
-          # other workflows
+          # Pass a personal access token (using our `ct-changesets` app) to be able to trigger other workflows
```
Comment (CarlosCortizasCT) [inline .github/workflows/release.yml:26]: Maybe I'm wrong but are we still using a PAT?

Comment (tdeekens) [reply .github/workflows/release.yml:26]: Yes, we're A using a [personal access token](https://help.github.com/en/github/authenticating-to-github/creating-a-personal-access-token-for-the-command-line) from a [personal user/bot account](https://help.github.com/en/github/getting-started-with-github/types-of-github-accounts#personal-user-accounts) in a 🥜shell.

**Comment by emmenko** [comment]
Thanks!

Also if you can clean up the repository secret that would be great 🤗

**Comment by tdeekens** [comment]
> Thanks!
> Also if you can clean up the repository secret that would be great 🤗

As I am doing this across many repos I just have a reminder for Monday to remove the PATs across all repos.

**Pull request merged**